### PR TITLE
Added SalesforceApi library

### PR DESCRIPTION
**Pull request by stuartmccoll**
## Proposed changes

This commit adds a `SalesforceApi` library, enabling creation of a `Restforce` client and retrieval of payment-related details for a funding application.

## Further comments

N/A.

## Screenshot (if applicable)

N/A.

## Checklist

- [X] Added tests for new functionality, or updated existing tests
- [X] Updated `README.md` if necessary

### CODE REVIEW

File: lib/apis/salesforce/salesforce_api.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    #
+    # @return [Hash] A Hash, currently containing only the amount awarded to the project
+    #                and the percentage of the total costs that the organisation have agreed
+    #                to award
+    def get_payment_related_details(id)
+
+      Rails.logger.info("Retrieving payment-related details for project ID: #{id}")
+
+      begin
+
```
Comment (ptrelease) [inline lib/apis/salesforce/salesforce_api.rb:30]: Does the below comment still apply?  Can't see grant % on line 31

File: lib/apis/salesforce/salesforce_api.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        api_version: '47.0'
+      )
+
+      Rails.logger.info('Finished initialising Salesforce client')
+
+    end
+
+  end
+
+end
```
Comment (ptrelease) [inline lib/apis/salesforce/salesforce_api.rb:86]: I like this.  I can see this module getting massive though, so at some point we may need to consider braking apart, or inheriting modules from another with the initialise_client method.

Comment (ptrelease) [reply lib/apis/salesforce/salesforce_api.rb:86]: How are you planning to instantiate and use salesforce_api.rb itself? Not something we need to worry about too much right now - but this implementation suggests you will implement as a singleton (which is what I'm hoping)

File: lib/apis/salesforce/salesforce_api.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        security_token: Rails.configuration.x.salesforce.security_token,
+        client_id: Rails.configuration.x.salesforce.client_id,
+        client_secret: Rails.configuration.x.salesforce.client_secret,
+        host: Rails.configuration.x.salesforce.host,
+        api_version: '47.0'
+      )
+
+      Rails.logger.info('Finished initialising Salesforce client')
+
+    end
```
Comment (ptrelease) [inline lib/apis/salesforce/salesforce_api.rb:82]: Todo: look up whether we need to close this connection

File: lib/apis/salesforce/salesforce_api.rb (hunk 1 above)
Comment (stuartmccoll) [reply lib/apis/salesforce/salesforce_api.rb:30]: Fixed in subsequent commit.

File: lib/apis/salesforce/salesforce_api.rb (hunk 2 above)
Comment (stuartmccoll) [reply lib/apis/salesforce/salesforce_api.rb:86]: Discussed offline.

File: lib/apis/salesforce/salesforce_api.rb (hunk 3 above)
Comment (stuartmccoll) [reply lib/apis/salesforce/salesforce_api.rb:82]: Discussed offline.

**Review by ptrelease — APPROVED** [review]

**Pull request merged**